src/examples/java/com/amazonaws/crypto/examples/keyring/awskms/ActLikeAwsKmsMasterKeyProvider.java

```diff
@@ -0,0 +1,143 @@
+ * <p>
+ * For more examples of how to use the AWS KMS keyring,
+ * see the 'keyring/awskms' directory.
+ */
+public class ActLikeAwsKmsMasterKeyProvider {
+
+    /**
+     * Demonstrate how to create a keyring that behaves like an AWS KMS master key provider.
+     *
+     * @param awsKmsCmk             The ARN of an AWS KMS CMK that protects data keys
+     * @param awsKmsAdditionalCmks  Additional ARNs of secondary AWS KMS CMKs
+     * @param sourcePlaintext       Plaintext to encrypt
+     */
+    public static void run(final AwsKmsCmkId awsKmsCmk, final List<AwsKmsCmkId> awsKmsAdditionalCmks, byte[] sourcePlaintext) {
+        // Instantiate the AWS Encryption SDK.
+        final AwsCrypto awsEncryptionSdk = new AwsCrypto();
+
+        // Prepare your encryption context.
+        // https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
+        final Map<String, String> encryptionContext = new HashMap<>();
+        encryptionContext.put("encryption", "context");
+        encryptionContext.put("is not", "secret");
+        encryptionContext.put("but adds", "useful metadata");
+        encryptionContext.put("that can help you", "be confident that");
+        encryptionContext.put("the data you are handling", "is what you think it is");
+
+        // This is the master key provider whose behavior we want to reproduce.
+        //
+        // When encrypting, this master key provider generates the data key using the first CMK in the list
+        // and encrypts the data key using all specified CMKs.
+        // However, when decrypting, this master key provider attempts to decrypt
+        // any data keys that were encrypted under an AWS KMS CMK.
+        final List<String> masterKeyProviderCmks = new ArrayList<>();
+        masterKeyProviderCmks.add(awsKmsCmk.toString());
+        masterKeyProviderCmks.addAll(awsKmsAdditionalCmks.stream().map(AwsKmsCmkId::toString).collect(toList()));
+        final KmsMasterKeyProvider masterKeyProviderToReplicate = KmsMasterKeyProvider.builder()
+                .withKeysForEncryption(masterKeyProviderCmks).build();
+
+        // Create a CMK keyring that encrypts and decrypts using the specified AWS KMS CMKs.
+        //
+        // This keyring reproduces the encryption behavior of the AWS KMS master key provider.
+        //
+        // The AWS KMS keyring requires that you explicitly identify the CMK
+        // that you want the keyring to use to generate the data key.
+        final Keyring cmkKeyring = StandardKeyrings.awsKmsBuilder()
+                .generatorKeyId(awsKmsCmk)
+                .keyIds(awsKmsAdditionalCmks)
+                .build();
+
+        // Create an AWS KMS discovery keyring that will attempt to decrypt
+        // any data keys that were encrypted under an AWS KMS CMK.
+        final Keyring discoveryKeyring = StandardKeyrings.awsKmsDiscoveryBuilder().build();
+
+        // Combine the single-CMK and discovery keyrings
```

Comment:
whoops; I missed this in the Python example too; I'll send out a PR to fix that

```suggestion
        // Combine the CMK and discovery keyrings
```